src/App.tsx

```diff
@@ -1,26 +1,251 @@
+  const [loading, setLoading] = useState(false);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentFilter, setCurrentFilter] = useState<FilterOptions>(
+    FilterOptions.All,
+  );
+  const [errorMessage, setErrorMessage] = useState('');
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [deletingId, setDeletingID] = useState([0]);
+  // const [selectedTodo, setSelectedTodo] = useState(false);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  const handleDelete = (todoId: number): Promise<void> => {
+    setDeletingID(prev => [...prev, todoId]);
+
+    return todoService
+      .deleteTodos(todoId)
+      .then(() => {
+        setLoading(true);
+        setTodos(prevTodos => prevTodos.filter(todo => todo.id !== todoId));
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+      })
+      .finally(() => {
+        setLoading(false);
+        setDeletingID((prev: number[]) => prev.filter(id => id !== todoId));
+        setTimeout(() => setErrorMessage(''), 3000);
+      });
+  };
+
+  const filteredTodos = useMemo(() => {
+    switch (currentFilter) {
+      case FilterOptions.Active:
+        return todos.filter(todo => !todo.completed);
+      case FilterOptions.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [currentFilter, todos]);
+
+  const newTodo: Todo = {
+    id: 0,
+    title: newTodoTitle.trim(),
+    userId: USER_ID,
+    completed: false,
+  };
+
+  const handleSubmit = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!newTodoTitle.trim()) {
+      setErrorMessage('Title should not be empty');
+      inputRef.current?.focus();
+      setTimeout(() => setErrorMessage(''), 3000);
+
+      return;
+    }
+
+    setTempTodo(newTodo);
+
+    try {
+      const addNewTodo = await todoService.postTodo(newTodo);
+
+      setTodos(prevTodos => [...prevTodos, addNewTodo]);
+      setNewTodoTitle('');
+      setErrorMessage('');
+    } catch {
+      setErrorMessage('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+      inputRef.current?.focus();
+      setTimeout(() => setErrorMessage(''), 3000);
+    }
+  };
+
+  const allCompleted = todos.length > 0 && todos.every(todo => todo.completed);
+
+  const activeTodos = todos.filter(todo => !todo.completed).length;
+
+  const completedAllTodos = () => {
+    const shouldCompleteAllTodos = !allCompleted;
+
+    setTodos(
+      todos.map(todo => ({ ...todo, completed: shouldCompleteAllTodos })),
+    );
+  };
+
+  const toggleTodoCompletion = (id: number) => {
+    setTodos(
+      todos.map(todo =>
+        todo.id === id ? { ...todo, completed: !todo.completed } : todo,
+      ),
+    );
+  };
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [todos, inputRef]);
+
+  function loadTodos() {
```

Comment:
It is important to follow the same way of creating functions everywhere
```suggestion
  const loadTodos = () => {
```